agent/envoyextensions/builtin/otel-access-logging/structs.go

```diff
@@ -0,0 +1,434 @@
+		resultErr = multierror.Append(resultErr, fmt.Errorf("failed to validate Config.%s: %w", field, err))
+	}
+
+	return resultErr
+}
+
+func (a *AccessLog) envoyGrpcService(cfg *cmn.RuntimeConfig) (*envoy_core_v3.GrpcService, error) {
+	target := a.GrpcService.Target
+	clusterName, err := a.getClusterName(cfg, target)
+	if err != nil {
+		return nil, err
+	}
+
+	var initialMetadata []*envoy_core_v3.HeaderValue
+	for _, meta := range a.GrpcService.InitialMetadata {
+		initialMetadata = append(initialMetadata, meta.toEnvoy())
+	}
+
+	return &envoy_core_v3.GrpcService{
+		TargetSpecifier: &envoy_core_v3.GrpcService_EnvoyGrpc_{
+			EnvoyGrpc: &envoy_core_v3.GrpcService_EnvoyGrpc{
+				ClusterName: clusterName,
+				Authority:   a.GrpcService.Authority,
+			},
+		},
+		Timeout:         target.timeoutDurationPB(),
+		InitialMetadata: initialMetadata,
+	}, nil
+}
+
+// getClusterName returns the name of the cluster for the OpenTelemetry access logging service.
+// If the extension is configured with an upstream OpenTelemetry access logging service then the name of the cluster for
+// that upstream is returned. If the extension is configured with a URI, the only allowed host is `localhost`
+// and the extension will insert a new cluster with the name "local_access_log", so we use that name.
+func (a *AccessLog) getClusterName(cfg *cmn.RuntimeConfig, target *Target) (string, error) {
+	var err error
+	clusterName := LocalAccessLogClusterName
+	if target.isService() {
+		if clusterName, err = target.clusterName(cfg); err != nil {
+			return "", err
+		}
+	}
+	return clusterName, nil
+}
+
+func (a *AccessLog) isGRPC() bool {
+	return a.GrpcService != nil
+}
+
+// toEnvoyCluster returns an Envoy cluster for connecting to the OpenTelemetry access logging service.
+// If the extension is configured with the OpenTelemetry access logging service locally via the URI set to localhost,
+// this func will return a new cluster definition that will allow the proxy to connect to the OpenTelemetry access logging
+// service running on localhost on the configured port.
+//
+// If the extension is configured with the OpenTelemetry access logging service as an upstream there is no need to insert
+// a new cluster so this method returns nil.
+func (a *AccessLog) toEnvoyCluster(_ *cmn.RuntimeConfig) (*envoy_cluster_v3.Cluster, error) {
+	target := a.GrpcService.Target
+
+	// If the target is an upstream we do not need to create a cluster. We will use the cluster of the upstream.
+	if target.isService() {
+		return nil, nil
+	}
+
+	host, port, err := target.addr()
+	if err != nil {
+		return nil, err
+	}
+
+	clusterType := &envoy_cluster_v3.Cluster_Type{Type: envoy_cluster_v3.Cluster_STATIC}
+	if host == localhost {
+		// If the host is "localhost" use a STRICT_DNS cluster type to perform DNS lookup.
+		clusterType = &envoy_cluster_v3.Cluster_Type{Type: envoy_cluster_v3.Cluster_STRICT_DNS}
+	}
+
+	var typedExtProtoOpts map[string]*anypb.Any
+	if a.isGRPC() {
```

Comment:
We don't need to do this check because OTEL is always over gRPC. We always need to insert the protocol options below.

Reply:
Removed the check